src/tespy/components/piping/pipeline.py

```diff
@@ -0,0 +1,378 @@
+                + 1 / (
+                    1 - (2 * self.pipe_depth.val / diameters[0]) ** 2
+                    * (1 + Beta) / (1 - Beta)
+                )
+            )
+            R_soil = (
+                _h / (2 * math.pi * lambda_ground[self.environment_media.val])
+            )
+
+            return (
+                self.inl[0].m.val_SI
+                * (self.outl[0].h.val_SI - self.inl[0].h.val_SI)
+                - dTm  / R_soil
+            )
+
+        else:
+            raise ValueError('No valid environmental media.')
+        if len(R_sum) == 0:
+            raise ValueError("No heat transfer resistance. Check input values.")
+
+        return (
+            self.inl[0].m.val_SI
+            * (self.outl[0].h.val_SI - self.inl[0].h.val_SI)
+            - dTm * area / sum(R_sum)
+        )
+
+    def _pipe_tc(self, material: str, T: float) -> float:
+        """Determine heat conduction of pipe material and temperature
+        Function from DWSIM:
+        https://github.com/DanWBR/dwsim/blob/windows/DWSIM.UnitOperations/UnitOperations/Pipe.vb#L1370-L1405
+
+        """
+        lamda = 0.0
+        if material in ["AoComum", "Steel"]:
+            lamda = (-0.000000004 * T ** 3) - (0.00002 * T ** 2) + (0.021 * T) + 33.743
+        elif material in ["AoCarbono", "CarbonSteel", "Carbon Steel"]:
+            lamda = (0.000000007 * T ** 3) - (0.00002 * T ** 2) - (0.0291 * T) + 70.765
+        elif material in ["FerroBottomido", "CastIron", "Cast Iron"]:
+            lamda = (-0.00000008 * T ** 3) + (0.0002 * T ** 2) - (0.211 * T) + 127.99
+        elif material in ["AoInoxidvel", "StainlessSteel", "Stainless Steel"]:
+            lamda = 14.6 + 0.0127 * (T - 273.15)
+        elif material in ["PVC", "PVC+PFRV"]:
+            lamda = 0.16
+        elif material in ["CommercialCopper", "CommercialCopper"]:
+            lamda = 420.75 - 0.068493 * T
+
+        return lamda  # W/m.K
+
+    def ohc_group_deriv(self, increment_filter, k):
+        """Calculate the partial derivatives of the ohc equation
+
+        Parameters
+        ----------
+        increment_filter : ndarray
+            Matrix for filtering non-changing variables.
+
+        k : int
+            Position of derivatives in Jacobian matrix (k-th equation).
+        """
+        i = self.inl[0]
+        o = self.outl[0]
+        func= self.ohc_group_func
+        if i.m.is_var:
+            self.jacobian[k, i.m.J_col] = o.h.val_SI - i.h.val_SI
+        if i.h.is_var:
+            self.jacobian[k, i.h.J_col] = self.numeric_deriv(func, 'h', i)
+        if o.h.is_var:
+            self.jacobian[k, o.h.J_col] = self.numeric_deriv(func, 'h', o)
+        if i.p.is_var:
+            self.jacobian[k, i.p.J_col] = self.numeric_deriv(func, 'p', i)
+        if o.p.is_var:
+            self.jacobian[k, o.p.J_col] = self.numeric_deriv(func, 'p', o)
+        if self.D.is_var:
+            self.jacobian[k, self.D.J_col] = self.numeric_deriv(func, 'D', None)
```

Comment:
I added the diameter as potential variable. I can see, that there is a case for identifying the pipe diameter while simultaneously imposing a pressure drop and a heat loss to the ambient.